includes/cli/class-ras-esp-sync.php

```diff
@@ -0,0 +1,376 @@
+				}
+			}
+		}
+
+		return static::$results['processed'];
+	}
+
+	/**
+	 * Does the given user have any subscriptions with an active status?
+	 *
+	 * @param int $user_id User ID.
+	 *
+	 * @return bool
+	 */
+	private static function user_has_active_subscriptions( $user_id ) {
+		$subcriptions = array_reduce(
+			array_keys( \wcs_get_users_subscriptions( $user_id ) ),
+			function( $acc, $subscription_id ) {
+				$subscription = \wcs_get_subscription( $subscription_id );
+				if ( $subscription->has_status( [ 'active', 'pending', 'pending-cancel' ] ) ) {
+					$acc[] = $subscription_id;
+				}
+				return $acc;
+			},
+			[]
+		);
+
+		return ! empty( $subcriptions );
+	}
+
+	/**
+	 * Get a batch of migrated subscriptions.
+	 *
+	 * This method requires the Newspack_Subscription_Migrations plugin to be
+	 * installed and active, otherwise it will return a WP_Error.
+	 *
+	 * @param string $source The source of the subscriptions. One of 'stripe', 'piano-csv', 'stripe-csv'.
+	 * @param int    $batch_size Number of subscriptions to get.
+	 * @param int    $offset Number to skip.
+	 * @param bool   $active_only Whether to get only active subscriptions.
+	 *
+	 * @return array|\WP_Error Array of subscription IDs or WP_Error.
+	 */
+	private static function get_migrated_subscriptions( $source, $batch_size, $offset, $active_only ) {
+		if (
+			! class_exists( '\Newspack_Subscription_Migrations\Stripe_Sync' ) ||
+			! class_exists( '\Newspack_Subscription_Migrations\CSV_Importers\CSV_Importer' )
+		) {
+			return new \WP_Error(
+				'newspack_esp_sync_contact',
+				__( 'The migrated-subscriptions flag requires the Newspack_Subscription_Migrations plugin to be installed and active.', 'newspack-plugin' )
+			);
+		}
+		$subscription_ids = [];
+		switch ( $source ) {
+			case 'stripe':
+				$subscription_ids = Stripe_Sync::get_migrated_subscriptions( $batch_size, $offset, $active_only );
+				break;
+			case 'piano-csv':
+				$subscription_ids = CSV_Importer::get_migrated_subscriptions( 'piano', $batch_size, $offset, $active_only );
+				break;
+			case 'stripe-csv':
+				$subscription_ids = CSV_Importer::get_migrated_subscriptions( 'stripe', $batch_size, $offset, $active_only );
+				break;
+			default:
+				return new \WP_Error(
+					'newspack_esp_sync_contact',
+					sprintf(
+						// Translators: %s is the source of the subscriptions.
+						__( 'Invalid subscription migration type: %s', 'newspack-plugin' ),
+						$source
+					)
+				);
+		}
+		return $subscription_ids;
+	}
+
+	/**
+	 * Get a batch of readers' IDs.
+	 *
+	 * @param int $batch_size Number of readers to get.
+	 * @param int $offset     Number to skip.
+	 *
+	 * @return array|false Array of user IDs, or false if no more to fetch.
+	 */
+	private static function get_batch_of_readers( $batch_size, $offset = 0 ) {
+		$roles = Reader_Activation::get_reader_roles();
+		if ( defined( 'NEWSPACK_NETWORK_READER_ROLE' ) && ! in_array( NEWSPACK_NETWORK_READER_ROLE, $roles, true ) ) {
```

Comment:
This is not necessary. Newspack Network, if active, will filter `get_reader_roles` and add its role there

Reply:
Well, another reason why this should've been reader roles all along 😄

Reply:
Fixed in e6253280f7a81f5284cde4b3d3d3e21f52fd2d00